### DISCUSSION TITLE
psychophysics & interval timing tasks

### DISCUSSION THREAD

**Discussion by benwandrew** [jspsych/jsPsych#1584] [Q&A]
hi all,

is there a standard plugin or preexisting ideas for coding psychophysical interval timing tasks in jspsych? i'm thinking particularly of comparative judgments of duration at the moment (eg, two sequential visual or audio stimuli presented, after which participant must select whether second was longer or shorter in duration), but also interested in interval timing more broadly. 

thanks very much!

**Comment by jodeleeuw** [comment]
Hi @benwandrew,

There's no plugin for this, but you could easily create such a task by combining plugins into a timeline and using timeline variables to set the duration for different trials. Here's a quick example:

```js

var interval_timing_task = {
	timeline: [
		{
			type: 'image-keyboard-response',
			stimulus: 'stim.png',
			choices: jsPsych.NO_KEYS,
			trial_duration: jsPsych.timelineVariables('first_duration'),
			post_trial_gap: 1000
		},
		{
			type: 'image-keyboard-response',
			stimulus: 'stim.png',
			choices: jsPsych.NO_KEYS,
			trial_duration: jsPsych.timelineVariables('second_duration'),
			post_trial_gap: 1000
		},
		{
			type: 'html-button-response',
			stimulus: '<p>Which image was on the screen longer?</p>',
			choices: ['First', 'Second'],
			data: {
				correct_answer: jsPsych.timelineVariable('correct_answer')
			}
		}
	],
	timeline_variables: [
		{first_duration: 500, second_duration: 600, correct_answer: 2},
		{first_duration: 600, second_duration: 500, correct_answer: 1},
		{first_duration: 500, second_duration: 800, correct_answer: 1},
		{first_duration: 800, second_duration: 500, correct_answer: 2}
	]
}
```
